### PR TITLE
removed a few hundred unused vars.

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

File: src/client/arakoon_remote_client_test.ml [hunk 1]
```diff
@@ -29,7 +29,7 @@ let _CLUSTER = "sweety"
 
 type real_test = Arakoon_client.client -> unit Lwt.t
 
-let __client_server_wrapper__ cluster (real_test:real_test) =
+let __client_server_wrapper__ _cluster (real_test:real_test) =
```
Comment (NicolasT) [inline src/client/arakoon_remote_client_test.ml]: Consider removing the argument instead?

File: src/node/catchup.ml [hunk 2]
```diff
@@ -105,7 +105,7 @@ let stop_fuse stop =
   else
     Lwt.return ()
 
-let catchup_tlog (type s) ~tls_ctx ~stop me other_configs ~cluster_id  mr_name ((module S : Store.STORE with type t = s),store,tlog_coll)
+let catchup_tlog (type s) ~tls_ctx ~stop _me other_configs ~cluster_id  mr_name ((module S : Store.STORE with type t = s),store,tlog_coll)
```
Comment (NicolasT) [inline src/node/catchup.ml]: Consider removing the argument instead, if the interface permits?

File: src/node/node_main.ml [hunk 3]
```diff
@@ -651,7 +651,7 @@ let _main_2 (type s)
       let unlock_killswitch (_:int) = Lwt_mutex.unlock killswitch in
       let listen_for_signal () = Lwt_mutex.lock killswitch in
 
-      let start_backend stop (master, constants, buffers, new_i, store) =
+      let start_backend stop (master, constants, buffers, new_i, _store) =
```
Comment (NicolasT) [inline src/node/node_main.ml]: Consider pruning the tuple?

File: src/node/sync_backend.ml [hunk 4]
```diff
@@ -243,7 +242,7 @@ struct
         self # _check_interval [key] >>= fun () ->
         let () = assert_value_size value in
         let update = Update.Set(key,value) in
-        let update_sets (update_result:Store.update_result) = Statistics.new_set _stats key value start in
+        let update_sets (_update_result:Store.update_result) = Statistics.new_set _stats key value start in
```
Comment (NicolasT) [inline src/node/sync_backend.ml:245]: Consider turning into `()`?

File: src/node/sync_backend.ml [hunk 5]
```diff
@@ -291,11 +290,13 @@ struct
         _update_rendezvous self update no_stats push_update ~so_post
 
       method aSSert ~consistency (key:string) (vo:string option) =
+        ignore consistency;
```
Comment (NicolasT) [inline src/node/sync_backend.ml]: I'm a bit surprised this can be ignored.

File: src/node/sync_backend.ml [hunk 6]
```diff
@@ -323,7 +324,7 @@ struct
           | Some w -> assert_value_size w
         in
         let update = Update.Replace(key,wanted) in
-        let update_stats ur = Statistics.new_replace _stats start in
+        let update_stats __ur = Statistics.new_replace _stats start in
```
Comment (NicolasT) [inline src/node/sync_backend.ml]: Typo?

File: src/node/sync_backend.ml [hunk 7]
```diff
@@ -350,10 +352,10 @@ struct
       method delete key = log_o self "delete %S" key >>= fun () ->
         let start = Unix.gettimeofday () in
         let update = Update.Delete key in
-        let update_stats ur = Statistics.new_delete _stats start in
+        let update_stats __ur = Statistics.new_delete _stats start in
```
Comment (NicolasT) [inline src/node/sync_backend.ml]: Typo?

File: src/nursery/routing.ml [hunk 8]
```diff
@@ -117,7 +117,7 @@ module Routing = struct
 *)
 
   let remove t c_id =
-    let rec inner_remove t c_id other_side =
+    let rec inner_remove t c_id _other_side =
```
Comment (NicolasT) [inline src/nursery/routing.ml]: Consider removing the argument?

File: src/tools/crash_logger.ml [hunk 9]
```diff
@@ -31,7 +31,7 @@ type e = {
 let circ_buf = Lwt_sequence.create()
 let msg_cnt = ref 0
 
-let add_to_crash_log section level msgs =
+let add_to_crash_log _section level msgs =
```
Comment (NicolasT) [inline src/tools/crash_logger.ml]: Consider removing the argument?

File: src/node/sync_backend.ml (hunk 5 above)
Comment (toolslive) [inline src/node/sync_backend.ml]: it's an update .... but the api needs to remain as is.

File: src/node/sync_backend.ml (hunk 6 above)
Comment (toolslive) [inline src/node/sync_backend.ml]: does it matter?

File: src/tools/crash_logger.ml (hunk 9 above)
Comment (toolslive) [inline src/tools/crash_logger.ml]: For a lot of these, there is a _feeling_ of an api. ie the log api has a section a level and msgs.... I would like to keep the signature, even though we don't use the parameters.

File: src/node/sync_backend.ml (hunk 5 above)
Comment (domsj) [inline src/node/sync_backend.ml]: Why does the API need to remain as is?

File: src/tools/crash_logger.ml (hunk 9 above)
Comment (domsj) [inline src/tools/crash_logger.ml]: It might make sense to dump the section to the crash log too?

File: src/node/node_cfg.ml [hunk 10]
```diff
@@ -405,7 +404,7 @@ module Node_cfg = struct
     let tlx_dir =
       let rec _find = function
         | [] -> tlog_dir
-        | x :: xs -> try get_string "tlf_dir"
+        | _ :: xs -> try get_string "tlf_dir"
```
Comment (domsj) [inline src/node/node_cfg.ml]: This non-usage actually indicates a bug

File: src/node/node_main.ml [hunk 11]
```diff
@@ -303,7 +303,7 @@ module X = struct
       Lwt.return r
     end
 
-  let on_accept (type s) statistics (tlog_coll:Tlogcollection.tlog_collection) (module S : Store.STORE with type t = s) store (v,n,i) =
+  let on_accept (type s) statistics (tlog_coll:Tlogcollection.tlog_collection) (module S : Store.STORE with type t = s) _store (v,n,i) =
```
Comment (domsj) [inline src/node/node_main.ml]: store and (module S) arg can be removed instead

File: src/node/sync_backend.ml (hunk 5 above)
Comment (toolslive) [inline src/node/sync_backend.ml]: ok, seems to be consensus the parameter can be removed from the signature, and it will remain in the protocol only.

Comment (domsj) [inline src/node/sync_backend.ml]: On second thought I think the api should remain as is, but we should no longer consider the operation to be an update but treat it as a query instead (and depending on ~consistency maybe allow the query to happen on a slave too).
Same for assert_exists below

Comment (toolslive) [inline src/node/sync_backend.ml]: ok. will leave it as is; turning this into a query can be done in a different PR.

**Comment by domsj** [comment]
LGTM

**Pull request merged**